src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1267,15 +1267,15 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
       original(tp1.appliedTo(tp1.typeParams.map(_.paramBoundsAsSeenFrom(tp1))), tp2)
     else
       PolyType(
-        paramNames = tpnme.syntheticLambdaParamNames(tparams1.length),
+        paramNames = tpnme.syntheticTypeParamNames(tparams1.length),
         variances = (tparams1, tparams2).zipped.map((tparam1, tparam2) =>
           (tparam1.paramVariance + tparam2.paramVariance) / 2))(
         paramBoundsExp = tl => (tparams1, tparams2).zipped.map((tparam1, tparam2) =>
           tl.lifted(tparams1, tparam1.paramBoundsAsSeenFrom(tp1)).bounds &
-          tl.lifted(tparams2, tparam2.paramBoundsAsSeenFrom(tp2)).bounds),
+            tl.lifted(tparams2, tparam2.paramBoundsAsSeenFrom(tp2)).bounds),
```

Comment:
Note to self: the formatting changes where done in 3ac15f0ec0fcdbe0aacd3952ec113a71a9910af2